### PR TITLE
feat(Sound): Add duration and processed event, visual test migration

### LINKED ISSUE #1476: Sounds do not stop playing when game is stopped
### Steps to Reproduce

1. Add sound
2. Play sound (loop for effect)
3. Call `game.stop()`

### Expected Result

The game should stop playing _and_ subsequent calls to `mySound.play()` should not begin playing.

### Actual Result

The sound keeps playing and you can keep calling `mySound.play()` to play more.

### Environment

- browsers and versions: All
- operating system: Windows
- Excalibur versions: 0.23.0
- (anything else that may be relevant) 

### Current Workaround

Manually stop all sounds.

### PR DESCRIPTION
**Pull request by kamranayub**
===:clipboard: PR Checklist :clipboard:===

- [x] :pushpin: issue exists in github for these changes
- [x] :microscope: existing tests still pass
- [x] :see_no_evil: code conforms to the [style guide](https://github.com/excaliburjs/Excalibur/blob/master/STYLEGUIDE.md)
- [x] :triangular_ruler: new tests written and passing / old tests updated with new scenario(s)
- [x] :page_facing_up: changelog entry added (or not needed)

==================

Closes #1476

## Changes:

- Fixes #1476: Ensure all sounds stop when game is stopped and `Sound.play()` respects when game is in stopped state
- Added: Add `duration` to `Sound` and `AudioInstance`, when using `AudioBuffer`
- Added: Add `processed` event to `Sound` that receives processed data (`string | AudioBuffer`)
- Changed: Change `Sound.stop()` to always stop and rewind the track, even if sound is not playing
- Add `wireEngine` test coverage in addition to new changes
- Migrate audio visual tests to Storybook
  - Uses Campire and Forest 1 loops from [PremiumBeat](https://www.premiumbeat.com/blog/free-ambient-background-tracks/)

### CODE REVIEW

File: src/engine/files.d.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@ declare module '*.png' {
   export default value;
 }
 
+declare module 'guitarLoop';
```
Comment (eonarheim) [inline src/engine/files.d.ts:6]: This seems odd to have in the src/engine files.d.ts, can we move this another place?

Comment (eonarheim) [reply src/engine/files.d.ts:6]: Perhaps a files.d.ts in the specific story root that can be referenced?

File: src/engine/Resources/Sound/Sound.ts [hunk 2]
```diff
@@ -166,13 +186,9 @@ export class Sound extends Resource<Blob | ArrayBuffer> implements Audio {
   }
 
   /**
-   * Stop the sound and rewind
+   * Stop the sound if it is currently playing and rewind the track. If the sound is not playing, rewinds the track.
```
Comment (eonarheim) [inline src/engine/Resources/Sound/Sound.ts:189]: 👍

File: package.json [hunk 3]
```diff
@@ -46,6 +46,7 @@
   },
   "devDependencies": {
     "@babel/core": "7.9.0",
+    "@fortawesome/fontawesome-free": "^5.13.0",
```
Comment (eonarheim) [inline package.json:49]: Pin to exact version 
```suggestion
    "@fortawesome/fontawesome-free": "5.13.0",
```

File: src/engine/files.d.ts (hunk 1 above)
Comment (kamranayub) [reply src/engine/files.d.ts]: This was a F2 rename-refactor issue, fixed.

**Comment by kamranayub** [comment]
I can look into why the Appveyor test is breaking 🤔

**Comment by kamranayub** [comment]
@eonarheim Looks like appveyor being appveyor, a rebuild worked.

**Pull request merged**